plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -68,6 +69,167 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
 
     // TODO move common tests from connector-specific classes here
 
+    @Test
+    public void testTopNPushdownDisabled()
+    {
+        // If topN pushdown gets enabled by default, this test should use a session with topN pushdown disabled
+        assertThat(query("SELECT orderkey FROM orders ORDER BY orderkey limit 10"))
+                .ordered()
+                .isNotFullyPushedDown(TopNNode.class);
+    }
+
+    /**
+     * Verify !SUPPORTS_TOPN_PUSHDOWN declaration is true.
+     */
+    @Test
+    public void verifySupportsTopNPushdownDeclaration()
+    {
+        if (hasBehavior(SUPPORTS_TOPN_PUSHDOWN)) {
+            // Covered by testTopNPushdown
+            return;
+        }
+
+        assertThat(query(topNPushdownEnabled(getSession()), "SELECT orderkey FROM orders ORDER BY orderkey limit 10"))
+                .ordered()
+                .isNotFullyPushedDown(TopNNode.class);
+    }
+
+    @Test
+    public void testTopNPushdown()
+    {
+        if (!hasBehavior(SUPPORTS_TOPN_PUSHDOWN)) {
+            assertThat(query("SELECT orderkey FROM orders ORDER BY orderkey limit 10"))
```

Comment:
It seems to me that you should just use `pushd-down-enabled` session here and then you can drop `verifySupportsTopNPushdownDeclaration`.
I guess I am missing something.

Reply:
Should be possibel. The style was copied over from join pushdown tests so @findepi and you would have more context why.